clients/rest-client-core/src/main/java/uk/gov/justice/services/clients/core/RestClientProcessor.java

```diff
@@ -72,7 +81,14 @@ public JsonEnvelope request(final EndpointDefinition definition, final JsonEnvel
 
         final Invocation.Builder builder = target.request(format(MEDIA_TYPE_PATTERN, envelope.metadata().name()));
 
+        final WebTarget finalTarget = target;
+        trace(LOGGER, () -> String.format("Sending REST request to %s using message: %s", finalTarget.getUri().toString(),
+                toEnvelopeTraceString(envelope)));
+
         final Response response = builder.get();
+
+        trace(LOGGER, () -> String.format("REST response received: %s", toResponseTrace(response)));
```

Comment:
We need to be able to tie this to the previous log messaging somehow, otherwise we might not know which request the response is for. Maybe make it "REST response for %s received: %s", where the first %s is the id of the request envelope.